File: ios/PlantPal/PlantPal/Models/Decoration.swift
import Foundation
import SwiftData

struct DecorationItem: Identifiable, Hashable {
    let id: String
    let name: String
    let category: DecorationCategory
    let assetName: String
    let cost: Int
    let isUnlockedByDefault: Bool
    
    static let allItems: [DecorationItem] = [
        .init(id: "pot_default", name: "陶土盆", category: .pot, assetName: "pot_default", cost: 0, isUnlockedByDefault: true),
        .init(id: "pot_ceramic", name: "陶瓷盆", category: .pot, assetName: "pot_ceramic", cost: 10, isUnlockedByDefault: false),
        .init(id: "pot_wooden", name: "木桶盆", category: .pot, assetName: "pot_wooden", cost: 15, isUnlockedByDefault: false),
        .init(id: "pot_golden", name: "金盆", category: .pot, assetName: "pot_golden", cost: 50, isUnlockedByDefault: false),
        .init(id: "pot_crystal", name: "水晶盆", category: .pot, assetName: "pot_crystal", cost: 80, isUnlockedByDefault: false),
        
        .init(id: "bg_garden", name: "花园", category: .background, assetName: "bg_garden", cost: 0, isUnlockedByDefault: true),
        .init(id: "bg_forest", name: "森林", category: .background, assetName: "bg_forest", cost: 20, isUnlockedByDefault: false),
        .init(id: "bg_beach", name: "海滩", category: .background, assetName: "bg_beach", cost: 20, isUnlockedByDefault: false),
        .init(id: "bg_night", name: "星空", category: .background, assetName: "bg_night", cost: 30, isUnlockedByDefault: false),
        .init(id: "bg_rainbow", name: "彩虹", category: .background, assetName: "bg_rainbow", cost: 50, isUnlockedByDefault: false),
        
        .init(id: "outfit_default", name: "默认", category: .outfit, assetName: "outfit_default", cost: 0, isUnlockedByDefault: true),
        .init(id: "outfit_crown", name: "小皇冠", category: .outfit, assetName: "outfit_crown", cost: 30, isUnlockedByDefault: false),
        .init(id: "outfit_scarf", name: "围巾", category: .outfit, assetName: "outfit_scarf", cost: 15, isUnlockedByDefault: false),
        .init(id: "outfit_glasses", name: "眼镜", category: .outfit, assetName: "outfit_glasses", cost: 20, isUnlockedByDefault: false),
        .init(id: "outfit_wings", name: "翅膀", category: .outfit, assetName: "outfit_wings", cost: 60, isUnlockedByDefault: false),
        .init(id: "outfit_party", name: "派对帽", category: .outfit, assetName: "outfit_party", cost: 25, isUnlockedByDefault: false),
    ]
    
    static func pots() -> [DecorationItem] { allItems.filter { $0.category == .pot } }
    static func backgrounds() -> [DecorationItem] { allItems.filter { $0.category == .background } }
    static func outfits() -> [DecorationItem] { allItems.filter { $0.category == .outfit } }
}

enum DecorationCategory: String, CaseIterable {
    case pot = "pot"
    case background = "background"
    case outfit = "outfit"
    
    var displayName: String {
        switch self {
        case .pot: return "花盆"
        case .background: return "背景"
        case .outfit: return "配饰"
        }
    }
    
    var icon: String {
        switch self {
        case .pot: return "🪴"
        case .background: return "🏞️"
        case .outfit: return "🎩"
        }
    }
}

@Model
class OwnedDecoration {
    var id: UUID = UUID()
    var itemId: String
    var purchasedAt: Date = Date()
    
    init(itemId: String) {
        self.itemId = itemId
    }
}

@Model
class PlayerWallet {
    var id: UUID = UUID()
    var coins: Int = 0
    
    init(id: UUID = UUID(), coins: Int = 0) {
        self.id = id
        self.coins = coins
    }
    
    static func createDefault() -> PlayerWallet {
        let wallet = PlayerWallet()
        wallet.coins = 0
        return wallet
    }
}

File: android/app/src/main/java/com/plantpal/model/Decoration.kt
package com.plantpal.model

data class DecorationItem(
    val id: String,
    val name: String,
    val category: DecorationCategory,
    val assetName: String,
    val cost: Int,
    val isUnlockedByDefault: Boolean
) {
    companion object {
        val allItems = listOf(
            DecorationItem("pot_default", "陶土盆", DecorationCategory.POT, "pot_default", 0, true),
            DecorationItem("pot_ceramic", "陶瓷盆", DecorationCategory.POT, "pot_ceramic", 10, false),
            DecorationItem("pot_wooden", "木桶盆", DecorationCategory.POT, "pot_wooden", 15, false),
            DecorationItem("pot_golden", "金盆", DecorationCategory.POT, "pot_golden", 50, false),
            DecorationItem("pot_crystal", "水晶盆", DecorationCategory.POT, "pot_crystal", 80, false),
            DecorationItem("bg_garden", "花园", DecorationCategory.BACKGROUND, "bg_garden", 0, true),
            DecorationItem("bg_forest", "森林", DecorationCategory.BACKGROUND, "bg_forest", 20, false),
            DecorationItem("bg_beach", "海滩", DecorationCategory.BACKGROUND, "bg_beach", 20, false),
            DecorationItem("bg_night", "星空", DecorationCategory.BACKGROUND, "bg_night", 30, false),
            DecorationItem("bg_rainbow", "彩虹", DecorationCategory.BACKGROUND, "bg_rainbow", 50, false),
            DecorationItem("outfit_default", "默认", DecorationCategory.OUTFIT, "outfit_default", 0, true),
            DecorationItem("outfit_crown", "小皇冠", DecorationCategory.OUTFIT, "outfit_crown", 30, false),
            DecorationItem("outfit_scarf", "围巾", DecorationCategory.OUTFIT, "outfit_scarf", 15, false),
            DecorationItem("outfit_glasses", "眼镜", DecorationCategory.OUTFIT, "outfit_glasses", 20, false),
            DecorationItem("outfit_wings", "翅膀", DecorationCategory.OUTFIT, "outfit_wings", 60, false),
            DecorationItem("outfit_party", "派对帽", DecorationCategory.OUTFIT, "outfit_party", 25, false)
        )

        fun pots() = allItems.filter { it.category == DecorationCategory.POT }
        fun backgrounds() = allItems.filter { it.category == DecorationCategory.BACKGROUND }
        fun outfits() = allItems.filter { it.category == DecorationCategory.OUTFIT }
    }
}

enum class DecorationCategory(val displayName: String, val icon: String) {
    POT("花盆", "🪴"),
    BACKGROUND("背景", "🏞️"),
    OUTFIT("配饰", "🎩")
}
